examples/ephemeral-spark/hbase-cluster.tf

```
@@ -37,20 +38,59 @@ module "emr-hbase" {
   emr_ec2_instance_profile_name = "${var.name_prefix}-hbase-emr-instance-profile"
   emr_service_iam_policy_name   = "${var.name_prefix}-hbase-service-policy"
   emr_ec2_iam_policy_name       = "${var.name_prefix}-hbase-ec2-policy"
-  master_instance_group_name    = "${var.name_prefix}-HBaseMasterInstanceGroup"
-  core_instance_group_name      = "${var.name_prefix}-HBaseCoreInstanceGroup"
-  emr_managed_master_sg_name    = "${var.name_prefix}-EMR-HBase-Master"
-  emr_managed_core_sg_name      = "${var.name_prefix}-EMR-HBase-Core"
-  emr_additional_master_sg_name = "${var.name_prefix}-EMR-HBase-Additional-Master"
-  emr_additional_core_sg_name   = "${var.name_prefix}-EMR-HBase-Additional-Core"
-  emr_service_access_sg_name    = "${var.name_prefix}-EMR-HBase-Service-Access"
+  master_instance_fleet_name    = "${var.name_prefix}-HBaseMasterInstanceGroup"
+  core_instance_fleet_name      = "${var.name_prefix}-HBaseCoreInstanceGroup"
+  emr_managed_sg_name           = "${var.name_prefix}-EMR-Managed"
 
   # Scale
-  master_group_instance_count = 1
-  core_group_instance_count   = 4
-  master_instance_type        = "m4.large"
-  core_instance_type          = "r5.4xlarge"
-  master_ebs_size             = 50
-  core_ebs_size               = 200
-  core_bid_price              = "1.260" # r5.4xlarge on emr -> $1.008 + $0.252 = $1.260
+  master_instance_on_demand_count = 1
+  core_instance_on_demand_count   = 4
+  # core_instance_spot_count    = 4
+  # core_bid_price_as_percentage_of_on_demand_price = 100
+  master_instance_type = "m4.large"
+  core_instance_type   = "r5.4xlarge"
```

Not your mistake, but r5.xlarge make a lot more sense for an hbase cluster.

HBase regions don't really make use of more than ~32GB of RAM.

```suggestion
  core_instance_type   = "r5.xlarge"
```